File: src/types/booking.ts
export type ServiceItem = {
  id: string
  nameTh: string
  nameEn: string
  descriptionTh: string
  durationMinutes: number
  priceCents: number
  accentColor: string
  isActive: boolean
}

export type AvailabilitySlot = {
  time: string
  booked: number
  capacity: number
  available: boolean
}

export type Booking = {
  id: string
  bookingCode: string
  serviceId: string
  service?: ServiceItem
  customerName: string
  phone: string
  lineUserId?: string
  notes?: string
  bookingDate: string
  slotTime: string
  status: 'pending' | 'confirmed' | 'completed' | 'cancelled' | 'no_show'
  createdAt: string
}

export type BookingBlackoutDate = {
  id?: string
  date: string
  reason?: string
}

export type BookingRules = {
  openTime: string
  closeTime: string
  slotIntervalMinutes: number
  slotCapacity: number
  closedWeekdays: string
  minAdvanceHours: number
  maxAdvanceDays: number
  reminderLeadMinutes: number
  bufferMinutes: number
  blackoutDates: BookingBlackoutDate[]
}

export type CreateBookingPayload = {
  serviceId: string
  customerName: string
  phone: string
  lineUserId?: string
  bookingDate: string
  slotTime: string
  notes?: string
}

export type RescheduleBookingPayload = {
  lineUserId: string
  bookingDate: string
  slotTime: string
  notes?: string
}

export type ApiEnvelope<T> = {
  data: T
}

File: apps/api/internal/models/booking.go
package models

const (
	BookingStatusPending   = "pending"
	BookingStatusConfirmed = "confirmed"
	BookingStatusCompleted = "completed"
	BookingStatusCancelled = "cancelled"
	BookingStatusNoShow    = "no_show"
)

type Booking struct {
	BaseModel
	BookingCode  string  `json:"bookingCode" gorm:"uniqueIndex;size:32;not null"`
	ServiceID    string  `json:"serviceId" gorm:"index;size:36;not null"`
	Service      Service `json:"service" gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT;"`
	CustomerName string  `json:"customerName" gorm:"size:180;not null"`
	Phone        string  `json:"phone" gorm:"size:40;not null"`
	LineUserID   string  `json:"lineUserId" gorm:"size:128;index"`
	Notes        string  `json:"notes" gorm:"size:1000"`
	BookingDate  string  `json:"bookingDate" gorm:"index;size:10;not null"`
	SlotTime     string  `json:"slotTime" gorm:"index;size:5;not null"`
	Status       string  `json:"status" gorm:"index;size:32;not null"`
	NoShowCount  int     `json:"noShowCount,omitempty" gorm:"-"`
}

type BookingFilter struct {
	Status string
	Date   string
	From   string
	To     string
	Query  string
	Limit  int
}

type BookingSettings struct {
	BaseModel
	OpenTime            string                `json:"openTime" gorm:"size:5;not null;default:'09:00'"`
	CloseTime           string                `json:"closeTime" gorm:"size:5;not null;default:'17:00'"`
	SlotIntervalMinutes int                   `json:"slotIntervalMinutes" gorm:"not null;default:30"`
	SlotCapacity        int                   `json:"slotCapacity" gorm:"not null;default:1"`
	ClosedWeekdays      string                `json:"closedWeekdays" gorm:"size:32;not null;default:''"`
	MinAdvanceHours     int                   `json:"minAdvanceHours" gorm:"not null;default:0"`
	MaxAdvanceDays      int                   `json:"maxAdvanceDays" gorm:"not null;default:60"`
	ReminderLeadMinutes int                   `json:"reminderLeadMinutes" gorm:"not null;default:1440"`
	BufferMinutes       int                   `json:"bufferMinutes" gorm:"not null;default:0"`
	BlackoutDates       []BookingBlackoutDate `json:"blackoutDates" gorm:"-"`
}

type BookingBlackoutDate struct {
	BaseModel
	Date   string `json:"date" gorm:"uniqueIndex;size:10;not null"`
	Reason string `json:"reason" gorm:"size:180"`
}
